src/server/pps/server/api_server.go

```diff
@@ -1455,6 +1569,133 @@ func (a *apiServer) runPipeline(ctx context.Context, pipelineInfo *ppsclient.Pip
 	}
 }
 
+func (a *apiServer) jobManager(ctx context.Context, job *ppsclient.Job) error {
```

Comment:
This could use a more descriptive name and maybe a comment about what this function does.